bus-mapping/src/evm/opcodes/mstore.rs

```diff
@@ -134,4 +148,77 @@ mod mstore_tests {
 
         Ok(())
     }
+
+    #[test]
+    fn mstore8_opcode_impl() -> Result<(), Error> {
+        let code = bytecode! {
+            .setup_state()
+            PUSH1(0x12)
+            PUSH2(0x100)
+            #[start]
+            MSTORE8
+            STOP
+        };
+
+        // Get the execution steps from the external tracer
+        let block =
+            mock::BlockData::new_single_tx_trace_code_at_start(&code).unwrap();
+
+        let mut builder = CircuitInputBuilder::new(
+            block.eth_block.clone(),
+            block.block_ctants.clone(),
+        );
+        builder.handle_tx(&block.eth_tx, &block.geth_trace).unwrap();
+
+        let mut test_builder = CircuitInputBuilder::new(
+            block.eth_block,
+            block.block_ctants.clone(),
+        );
+        let mut tx = Transaction::new(&block.eth_tx);
+        let mut tx_ctx = TransactionContext::new(&block.eth_tx);
+
+        // Generate step corresponding to MSTORE
+        let mut step = ExecStep::new(
+            &block.geth_trace.struct_logs[0],
+            0,
+            test_builder.block_ctx.gc,
+            0,
+        );
+        let mut state_ref =
+            test_builder.state_ref(&mut tx, &mut tx_ctx, &mut step);
+
+        // Add StackOps associated to the 0x100, 0x12 reads starting from last
+        // stack position.
+        state_ref.push_op(StackOp::new(
+            RW::READ,
+            StackAddress::from(1022),
+            Word::from(0x100),
+        ));
+        state_ref.push_op(StackOp::new(
+            RW::READ,
+            StackAddress::from(1023),
+            Word::from(0x12),
+        ));
+
+        // Add 1 MemoryOp generated from the Memory write at addr 0x100.
+        state_ref.push_op(MemoryOp::new(
+            RW::WRITE,
+            MemoryAddress(0x100),
+            *Word::from(0x12).to_le_bytes().first().unwrap(),
```

Comment:
Why not set the expected value directly?  Like just `0x12`.  Or if you follow my previous comment, `0x34`.